### PR TITLE
Fix Issue 19464 - typeof immutable fields order dependent

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -902,6 +904,7 @@ Lagain:
             e = new DotVarExp(loc, new ThisExp(loc), v);
         else
             e = new VarExp(loc, v);
+
```
Comment (thewilsonator) [inline src/dmd/expressionsem.d]: ?

**Review by thewilsonator — APPROVED** [review]
Otherwise  LGTM.

**Comment by RazvanN7** [comment]
@thewilsonator Done.

**Review by thewilsonator — APPROVED** [review]

**Comment by UplinkCoder** [comment]
@RazvanN7 maybe you should mention, in the commit message, what the bug was and why your change fixes it.

**Comment by JinShil** [comment]
According to a digger bisection, this PR may have introduced a regression: https://issues.dlang.org/show_bug.cgi?id=19661

**Pull request merged**